midend/README.md

```diff
@@ -1,3 +1,31 @@
+<!--!
+\page midend Midend                                                                      
+-->
+<!-- 
+Documentation Inclusion:
+This README is integrated as a standalone page in the P4 compiler documentation.
+
+Refer to the full page here: [Midend](https://p4lang.github.io/p4c/midend.html)
+-->
+<!--!
+\internal
+-->
+# Midend
+
 This folder contains passes which may be useful for implementing various mid-ends.
 These are linked as part of the front-end library.
 These passes are not used in the front-end, but are still largely architecture-independent.
+
+<!--!
+\endinternal
+-->
+The mid-end stage applies further optimizations to the IR of the P4 program. It's architecture-independent, it is driven by policies specific to the target architecture. The mid-end uses the same base IR as the front-end and includes the following passes:
+
+## Optimization Passes:
```

Comment:
This seems incomplete? Maybe hold this back or remove it until we have a full list (which is totally optional).

Reply:
Remove the whole description ?

Reply:
I would, it's a big ask to add descriptions to all passes.

Reply:
Should I remove the section or leave the points and remove the description.